### PR TITLE
Make original filename available in the API

### PR DESCRIPTION
**Pull request by jorgemmsilva**

### CODE REVIEW

File: src/Upload/index.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ export default class Upload extends Component {
             .forEach(
                 (file) => {
                     let reader = new FileReader();
+                    reader.fileName = file.name
```
Comment (corpix) [inline src/Upload/index.js]: Why we should define a custom property on `FileReader`? Why not wrap the `this.props.onFileLoad` like this?
``` javascript
reader.onload = (e) => this.props.onFileLoad(e, file);
```

**Review by corpix — CHANGES_REQUESTED** [review]

**Comment by jorgemmsilva** [comment]
so you can do something like this:
```
  onFileLoad = (e) => {
      this.setState({
        newFileData: e.target.result,
        newFileName: e.target.name,
      })
  }
```

**Comment by corpix** [comment]
It is ok to have a file name in the `onFileLoad` handler, but it is bad to define it directly on `FileReader` instance, you should not modify browser-providen things and things around. The more clean solution is to define additional argument for a handler. At this moment we have `e` in `onFileLoad`, now we need a file name too, so we could wrap `onFileLoad` in this way

``` javascript
reader.onload = (e) => this.props.onFileLoad(e, file);
```

So code in your example will look like this:

``` javascript
    onFileLoad = (e, file) => {
      this.setState({
        newFileData: e.target.result,
        newFileName: file.name,
      })
    }
```

**Comment by jorgemmsilva** [comment]
I was thinking about simplicity for the end user... But I do get your point about modifying native objects :p

updated

**Comment by corpix** [comment]
Thank you

**Pull request merged**